File: app/src/main/java/com/summerland/android/game/Player.java
package com.summerland.android.game;

import android.graphics.Bitmap;
import android.graphics.Canvas;
import android.graphics.Point;

/**
 * Player class, Created by steve on 01/10/16.
 */
public class Player extends GameObject {
    private int score = 0;
    private boolean goingUp = false, playing = false, collided = false;
    private Animation animation = new Animation();

    Player(Bitmap res, int posX, int posY, int sizeW, int sizeH, int numFrames) {
        x = posX;
        y = posY;
        dy = 0;

        spriteSheet = res;
        width = sizeW;
        height = sizeH;

        Bitmap[] image = new Bitmap[numFrames];

        for (int i = 0; i < numFrames; i ++) {
            image[i] = Bitmap.createBitmap(spriteSheet, i*width, 0, width, height);
        }

        animation = new Animation();
        animation.setFrames(image);
        animation.setDelay(ANIM_DELAY);
        startTime = System.nanoTime();
    }

    boolean isGoingUp() {
        return this.goingUp;
    }

    void setGoingUp(boolean goingUp) {
        this.goingUp = goingUp;
    }

    @Override
    public void update() {
        elapsed = System.nanoTime()-startTime/MS;
        if (elapsed > ELAPSED_MS) {
            score++;
            startTime = System.nanoTime();
            // System.out.println("Players Score: " +  score);
        }

        animation.update();

        if (goingUp) dy -= 1;
        else dy += 1;
        if (dy > 14) dy = Player.MAX_GRAVITY;
        if (dy < -14) dy = -Player.MAX_GRAVITY;

        y += dy*2;
    }

    @Override
    public void draw(Canvas canvas) {
        canvas.drawBitmap(animation.getImage(), x, y, null);
    }

    int getScore() {
        return score;
    }

    boolean isPlaying() {
        return playing;
    }

    void setPlaying(boolean playing) {
        if (playing) score = 0;
        this.playing = playing;
    }

    void reset(Point point) {
        x = point.x;
        y = point.y;
        //score = 0;
        dy = 0;
        goingUp = false;
        playing = false;
        collided = false;
    }

    boolean isCollided() {
        return collided;
    }

    void setCollided() {
        this.collided = true;
    }
}

File: app/src/main/java/com/summerland/android/game/Player.kt
package com.summerland.android.game

import android.graphics.Bitmap
import android.graphics.Canvas
import android.graphics.Point

/**
 * Player class, Created by steve on 01/10/16.
 */
class Player internal constructor(res: Bitmap, posX: Int, posY: Int, sizeW: Int, sizeH: Int, numFrames: Int) : GameObject() {
    internal var score = 0
        private set
    internal var isGoingUp = false
    private var playing = false
    internal var isCollided = false
        private set

    internal var isPlaying: Boolean
        get() = playing
        set(playing) {
            if (playing) score = 0
            this.playing = playing
        }

    init {
        x = posX
        y = posY
        dy = 0

        spriteSheet = res
        width = sizeW
        height = sizeH

        val image = arrayOfNulls<Bitmap>(numFrames)

        for (i in 0 until numFrames) {
            image[i] = Bitmap.createBitmap(spriteSheet!!, i * width, 0, width, height)
        }

        animation = Animation()
        animation!!.setFrames(image)
        animation!!.setDelay(GameObject.ANIM_DELAY.toLong())
        startTime = System.nanoTime()
    }

    override fun update() {
        elapsed = System.nanoTime() - startTime / GameObject.MS
        if (elapsed > GameObject.ELAPSED_MS) {
            score++
            startTime = System.nanoTime()
            // System.out.println("Players Score: " +  score);
        }

        animation!!.update()

        if (isGoingUp)
            dy -= 1
        else
            dy += 1
        if (dy > 14) dy = GameObject.MAX_GRAVITY
        if (dy < -14) dy = -GameObject.MAX_GRAVITY

        y += dy * 2
    }

    override fun draw(canvas: Canvas) {
        canvas.drawBitmap(animation!!.image, x.toFloat(), y.toFloat(), null)
    }

    internal fun reset(point: Point) {
        x = point.x
        y = point.y
        //score = 0;
        dy = 0
        isGoingUp = false
        playing = false
        isCollided = false
    }

    internal fun setCollided() {
        this.isCollided = true
    }
}
